source/slang/slang-ir-specialize-resources.cpp

```diff
@@ -1349,4 +1352,16 @@ bool isIllegalSPIRVParameterType(IRType* type, bool isArray)
     }
     return false;
 }
+
+bool isIllegalWGSLParameterType(IRType* type)
+{
+    // The WGSL emitter translates structured buffer types to runtime-sized array types.
+    // Functions cannot take parameters of runtime-sized array type in WGSL.
```

Comment:
It should be more than this. Basically every illegal glsl/spirv parameter type is also illegal wgsl parameter type.

We can simply have this function call into isIllegalGLSLParameterType for now.

Reply:
Ok, done. I'd personally rather just copy-paste the function in case these need to diverge at any point.

I don't think it's obvious that these two sets of IR types should be equal for GLSL and WGSL, in all present and future cases.